avro/src/main/scala/hydra/avro/convert/ISODateConverter.scala

```diff
@@ -0,0 +1,28 @@
+package hydra.avro.convert
+
+import java.time.{OffsetDateTime, ZoneId, ZonedDateTime}
+
+import org.apache.avro.{Conversion, LogicalType, Schema}
+
+class ISODateConverter extends Conversion[ZonedDateTime] {
+
+  override def getLogicalTypeName: String = IsoDate.IsoDateLogicalTypeName
+
+  override def getConvertedType: Class[ZonedDateTime] = classOf[ZonedDateTime]
+
+  override def fromCharSequence(value: CharSequence,
+                                schema: Schema, `type`: LogicalType): ZonedDateTime = {
+    val odt = OffsetDateTime.parse(value)
+    odt.toLocalDateTime.atZone(ZoneId.of("UTC"))
+  }
+}
+
+object IsoDate extends LogicalType("iso-date") {
```

Comment:
Minor nitpick maybe.  I'd suggest we go with "iso-datetime" rather than just "iso-date".  There is an iso date and it looks like this: "2010-01-25".  Datetime on the other hand needs to have the time component, i.e., "2010-01-25T16:42:25Z" or "2018-04-24T13:47:06+00:00".  Postgres has native types for date and timestamp (date and time).  I don't want users to be confused that they might be getting just the date.